packages/jupyter-ai/jupyter_ai/auth/identity.py

```diff
@@ -0,0 +1,30 @@
+import getpass
+
+from jupyter_server.auth.identity import IdentityProvider, User
+
+
+def create_initials(username):
+    """Creates initials combining first 2 consonants"""
+
+    username = username.lower()
+
+    # Default: return first two unique consonants
+    consonants = [c for c in username if c in "bcdfghjklmnpqrstvwxyz"]
+    if len(consonants) >= 2:
+        return (consonants[0] + consonants[1]).upper()
+
+    # Fallback: first two characters
+    return username[:2].upper()
+
+
+class LocalIdentityProvider(IdentityProvider):
+
+    def get_user(self, handler):
+        username = getpass.getuser()
```

Comment:
Note that `getpass.getuser()` throws `OSError` if user name is not set on the system (see https://docs.python.org/3/library/getpass.html#getpass.getuser). If this is not a desired behavior we should provide a fallback value.

Reply:
Good catch! 
I did not find a good way to handle this, my first option was to rely on the base class `get_user`, but I ran into a number of `coroutine` related issues calling that method. See this [commit](https://github.com/jupyterlab/jupyter-ai/pull/1333/commits/7fd95a9128797fc4926c4bfb80eddcf1d5f1ebe1) for that implementation.

The second option would be to replicate much of the logic in `_get_user`, but that seems counter productive as it will diverge the code when future changes happen in base identity provider.

I have added unit tests, but leaving this as is with an error message explaining that the identity provider failed and so users can switch to a different identity provider.

Reply:
Thank you for looking into it. Relying on base class `get_user` definitely seems like a potential optimal solution. 

I noticed that in base class implementation [`get_user` method just calls `_get_user`](https://github.com/jupyter-server/jupyter_server/blob/d72b1a27f6478de898783ed482b8c25402bd2ab7/jupyter_server/auth/identity.py#L222) which our class does not overwrite and therefore inherits. I tried your [commit](https://github.com/jupyterlab/jupyter-ai/pull/1333/commits/7fd95a9128797fc4926c4bfb80eddcf1d5f1ebe1) with `self._get_user(handler)` instead of `super().get_user(handler)` while throwing an `OSError` and it seems to work great.

Reply:
Thanks for looking into this 👍. Updated as per your findings.

Reply:
Thank you.